docs/designs/CodeFlare-SDK-design-doc.md

```diff
@@ -0,0 +1,137 @@
+│       |   ├── kueue/
+│       |   │   ├── __init__.py
+│       |   │   ├── kueue.py
+│       |   │   └── test_kueue.py
+│       |   ├── widgets/
+│       |   │   ├── __init__.py
+│       |   │   ├── widgets.py
+│       |   │   └── test_widgets.py
+│       |   ├── utils/
+│       |   │   ├── __init__.py
+│       |   │   ├── utils.py
+│       |   │   └── test_utils.py
+│       ├── ray/
+│       │   ├── cluster/
+│       │   ├── client/
+│       │   ├── appwrapper/
+├── tests/
+│   ├── e2e/
+│   └── upgrade/
+├── poetry.lock
+└── pyproject.toml
+```
+
+### Client side best practices:
+1. The SDK should perform client-side validation of user inputs before sending them to the API server, to the extent possible. These validations should be limited to client-side checks, and the SDK is not responsible for performing validations that require server-side data. The SDK does not need to maintain the state of server-side objects. Examples of client-side validations include verifying user inputs, validating YAML schemas, and checking resource requests/limits based on available data.
+2. Any call made to the server should have a client instance passed through it. By default, the SDK should use the same client for all operations unless otherwise specified. For example, the client used to list clusters should also be used to fetch cluster details. The codebase should be designed so that these operations are either methods of the same client (e.g., a k8s client, RayJob client, TrainingClient) or the client instance is passed as an argument to the methods.
+3. Errors returned from the API server, to the extent possible, should be wrapped and handled explicitly.
+
+## Release:
+A new version of CodeFlare SDK will be released once every three weeks.
+For details on the release support matrix with other CodeFlare components, refer [here][codeflare_compatibility_matrix].
+RHOAI support matrix: CodeFlare SDK APIs fall under [Tier 2][RH_customer_API_support] support on `RHOAI` platform. This implies than an API would be support in for `n-1` major versions for a minimum of 9 months.
+
+## Security:
+Currently, users must manually manage mTLS certificates and re-authenticate Ray clients when connecting to Ray clusters, though the CodeFlare SDK offers helpers to streamline this process (as seen [here][cert_generation]). In the future, integration with a service mesh like Istio will automate mTLS setup and certificate management via a sidecar proxy, eliminating the need for manual certificate handling and client re-authentication.
+
+## Documentation and Examples:
+The following are the documentation sources for users which will need to remain updated:
+1. Upstream documentation - includes `README` and examples in `docs/` folder.
+2. Demo notebooks - The NBs which are to be tested and updated to ensure that they are upto date with the most recent changes.
+3. For RHOAI customers - [Official documentation][DW_RHOAI_docs] will be updated after every release.
```

Comment:
lets add ODH here too

Reply:
IIUC ODH support cycle would be the same as RHOAI. Will just cross-verify once again. Leaving the conversation open for now.

Reply:
Looks like it's the same. Have modified the statement.